### ISSUE TITLE
0.45.3 MeqHQ Corrupted Save, Ongoing Refit

### ISSUE THREAD

**Issue by OgraChief** [MegaMek/mekhq#1182]
Salutations,

I was directed to post my problem here from the Classic BT Forums. 

Environment: 
MekHQ 0.45.3
Jave Version 8 Update 201 (build 1.8.0_201-b)9)
Windows 10

Description:

So, I am trying to recover a saved game (campaign, megamekhq 42.2/45.3), and was hoping a solution can be found here. The save opens up fine, but when I get to the Hangar Bay it will not appear without some meddling. The problem appears to be related to a custom mek refit, a Dragon, DRG-1ZS; but maybe there is a deeper issue? Once I sell and/or GM delete the unit in question, the Hangar Bay works fine. The save appears to work correctly moving forward. However, anytime I advance the day, no matter the number of days, and save; I cannot reopen the file. I am prepared to lose units so long as the save can be salvaged. Any help would be appreciated.

Included files in a zip., if that is okay?

Thanks in advance.
[MeqHQ Corrupted Save, 13R.zip](https://github.com/MegaMek/mekhq/files/3091377/MeqHQ.Corrupted.Save.13R.zip)

**Comment by HammerGS** [comment]
Looks like two different NPE's in the MekHQ log.

First
```
MechBuster (Standard) #2 (self) CRIPPLED: only 0% internals remaining.
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at mekhq.campaign.parts.Refit.getDesc(Refit.java:1664)
	at mekhq.campaign.parts.Refit.partsInTransit(Refit.java:1031)
	at mekhq.campaign.parts.Refit.acquireParts(Refit.java:1048)
	at mekhq.campaign.Campaign.refit(Campaign.java:2375)
	at mekhq.campaign.Campaign.processNewDayUnits(Campaign.java:3059)
	at mekhq.campaign.Campaign.newDay(Campaign.java:3112)
	at mekhq.gui.CampaignGUI.advanceDay(CampaignGUI.java:1460)
	at mekhq.gui.CampaignGUI.access$2500(CampaignGUI.java:168)
	at mekhq.gui.CampaignGUI$53.actionPerformed(CampaignGUI.java:1278)
```
Second
```
MechBuster (Standard) #2 (self) CRIPPLED: only 0% internals remaining.
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at mekhq.campaign.parts.Refit.getDesc(Refit.java:1664)
	at mekhq.gui.model.UnitTableModel.getTooltip(UnitTableModel.java:163)
	at mekhq.gui.model.UnitTableModel$Renderer.getTableCellRendererComponent(UnitTableModel.java:308)
```

**Comment by HammerGS** [comment]
Have you tried this in the 45.4 release?

**Comment by OgraChief** [comment]
I just tried, and the file no longer opens. Attached are the logs for 0.45.4. Which appear to be similar to what I was receiving before. 
[0 45 4, megameklog.txt](https://github.com/MegaMek/mekhq/files/3091742/0.45.4.megameklog.txt)
[0 45 4, mekhqlog.txt](https://github.com/MegaMek/mekhq/files/3091743/0.45.4.mekhqlog.txt)

**Comment by OgraChief** [comment]
Just tried, and posted the logs for 0.45.4 release, on github. The campaign
file will not open.

I have no problem deleting units, just need to know which?

**Comment by VicenteCartas** [comment]
At least one of the problems seems related to the money changes. I'll give it a look.

**Comment by VicenteCartas** [comment]
The transist cost for one of the contracts is null, that's why it's failing to load. This will be fixed in the next version, thanks a lot for reporting it!

**Comment by VicenteCartas** [comment]
(About the refits, I'm not sure what's going on there)

**Comment by OgraChief** [comment]
It appears the initial problem was related to the refits after all. I simply restarted the refits and let them cycle to completion, probably could have deleted the offending refitted units as well, but restarting the ongoing refits worked for me. I also scrapped the above offending unit (mechbuster) as well. File now opens and works with 0.45.4, though still getting a number of errors in the logs. I will likely close (GM) the bugged contracts also and move on. So, the save is functional once more. If there are further suggestions, let me know. Just happy to have my save back. Thank you.
